### PR TITLE
feature: tgpt implementation

### PR DESCRIPTION
**Pull request by mattprintz**
# Description

squashed set of changes for TGPT implementation

Resolves #(issue)

### CODE REVIEW

**Review by dvince2 — APPROVED** [review]

File: docker-compose.yaml [hunk 1]
```diff
@@ -44,21 +49,51 @@ services:
     image: ghcr.io/darpa-askem/data-service:latest
     environment:
       DATASET_STORAGE_BASE_URL: s3://datasets/
+      STORAGE_HOST: http://minio:9000
```
Comment (dvince2) [inline docker-compose.yaml:52]: minio should be gone now with everything going through s3

File: docker-compose.yaml [hunk 2]
```diff
@@ -44,21 +49,51 @@ services:
...
+      S3_BUCKET: askem
+      S3_DATASET_PATH: datasets
+      S3_RESULT_PATH: simulations
```
Comment (dvince2) [inline docker-compose.yaml:55]: Will now also need `S3_ARTIFACT_PATH: artifacts`

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/TGPTResource.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	@ConfigProperty(name = "tgpt.baseUrl", defaultValue = "/chatty/")
+	String baseUrl;
+
+	@ConfigProperty(name = "tgpt.appUrl", defaultValue = "http://localhost:8078/chatty/")
+	String appUrl;
+
+	@ConfigProperty(name = "tgpt.wsUrl", defaultValue = "ws://localhost:8078/chatty_ws/")
+	String wsUrl;
+
+	@ConfigProperty(name = "tgpt.token", defaultValue = "89f73481102c46c0bc13b2998f9a4fce")
```
Comment (dvince2) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/TGPTResource.java:30]: Should this token be in the clear?

**Review by dvince2 — CHANGES_REQUESTED** [review]
Minio should be gone now as far as i know :o

File: docker-compose.yaml (hunk 1 above)
Comment (chris-dickson) [reply docker-compose.yaml:52]: This is for matt's docker-compose.  It's fine.

File: docker-compose.yaml (hunk 2 above)
Comment (chris-dickson) [reply docker-compose.yaml:55]: Don't worry about the docker-compose

File: docker-compose.yaml (hunk 1 above)
Comment (mattprintz) [reply docker-compose.yaml:52]: orchestration uses s3, but docker-compose doesn't. I don't have your S3 keys.
For now, at least, keeping minio in the docker compose.

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/TGPTResource.java (hunk 3 above)
Comment (chris-dickson) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/TGPTResource.java:30]: It's fine.  For local dev only.  There will a secret key pair in production that's encrypted.

File: docker-compose.yaml (hunk 2 above)
Comment (mattprintz) [reply docker-compose.yaml:55]: Probably true, but not worrying about it for this commit. Can add that later once this is done and we are doing something with artifacts via docker compose.

**Review by dvince2 — APPROVED** [review]

**Pull request merged**